Dockerfile

```diff
@@ -13,39 +13,61 @@
 
 FROM centos:7
 
-# Install xrootd
-RUN rpm -Uvh http://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
+RUN yum install -y git wget epel-release
+RUN yum group install -y "Development Tools"
 
-RUN yum install -y \
-    gcc \
-    python-devel \
-    xrootd \
-    xrootd-client \
-    xrootd-client-devel \
-    xrootd-python git \
-    xrootd-server \
+# Install and set python3.6 as default
+RUN yum install -y centos-release-scl && \
+    yum install -y rh-python36
+RUN echo '#!/bin/bash' >> /etc/profile.d/enablepython36.sh && \
+    echo '. scl_source enable rh-python36' >> /etc/profile.d/enablepython36.sh
+ENV BASH_ENV=/etc/profile.d/enablepython36.sh
 
-RUN yum install -y python-pip
+RUN chmod -R g=u /etc/profile.d/enablepython36.sh /opt/rh/rh-python36 && \
+    chgrp -R 0 /etc/profile.d/enablepython36.sh /opt/rh/rh-python36
+SHELL ["/bin/bash", "-c"]
 
-RUN adduser --uid 1001 xrootduser
+RUN yum-config-manager --enable rhel-server-rhscl-7-rpms
+RUN yum install -y devtoolset-7
+RUN scl enable devtoolset-7 bash
+
+# Install xrootd
+ARG xrootd_version=""
+ARG cmake="cmake3"
+RUN yum-config-manager --add-repo https://xrootd.slac.stanford.edu/binaries/xrootd-stable-slc7.repo
+RUN if [ ! -z "$xrootd_version" ] ; then XROOTD_V="-$xrootd_version" ; else XROOTD_V="" ; fi && \
+    echo "Will install xrootd version: $XROOTD_V (latest if empty)" && \
+    yum --setopt=obsoletes=0 install -y "$cmake" \
+                                        gcc-c++ \
+                                        zlib-devel \
+                                        openssl-devel \
+                                        libuuid-devel \
+                                        python3-devel \
+                                        xrootd"$XROOTD_V"
+
+RUN pip install --upgrade "pip<20" "setuptools<58" wheel
```

Comment:
Without pinning, `pip` resolvers never ends. This should be removed when upgrading everything with xrootd v5